### PR TITLE
Fix zoom reset

### LINKED ISSUE #1731: Zoom resets on dummy connect
Reproducible in 12.4.2
But probably it was always so =)

### PR DESCRIPTION
**Pull request by archimede67**
Fix zoom reset when connecting dummy.
This should fix #1731

### CODE REVIEW

**Comment by def-** [comment]
The main question is whether we want this behaviour.

**Comment by archimede67** [comment]
Yeah, but we can also ask why does it reset ?

**Comment by fokkonaut** [comment]
as i said in the issue, i am against this
Main reason is, that connecting dummy means a new player to play as, and that player should get the default zoomlevel

**Comment by archimede67** [comment]
Since it's your dummy i think you want to have the same zoom level as your main character when you connet it imo

**Comment by fokkonaut** [comment]
No, its a new player and should have the default zoom level, i see no reason to keep the zoom Level

**Comment by archimede67** [comment]
Because it resets the zoom even for your main character and that sucks

**Comment by 12pm** [comment]
then maybe change that?

**Comment by fokkonaut** [comment]
No, the zoom level should stay the same for both, just like it is. Its perfect

**Comment by archimede67** [comment]
Should we make a poll?

**Comment by def-** [comment]
I can play dictator and say let's merge this. :D

**Comment by def-** [comment]
The argument being that the camera setting should not be specific for the tee, but for the client. So it makes sense not to reset it when connecting dummy.
LGTM, let's merge when another admin says +1.

**Comment by BannZay** [comment]
@12pm, whats the point of having different zoomlevels on dummy? I find unconvinience only in this

@fokkonaut, so reset whole binds for dummy also?
It is never about "character"  it is about exactly "player". Distinguish those two, please. 

See nothing reasonable against

**Comment by fokkonaut** [comment]
does this set the zoom to 1 or to the default zoom level?

**Comment by fokkonaut** [comment]
@def- just merge it :)

**Comment by 12pm** [comment]
> @12pm, whats the point of having different zoomlevels on dummy? I find unconvinience only in this

e.g. spectating a big part of the map zoomed out on one tee and casually playing with the other one

**Comment by BannZay** [comment]
@12pm, everything seems to be nails when you learn what hammer is. 
Just use bindings in your case

File: src/game/client/components/camera.cpp [hunk 1]
```diff
@@ -29,8 +29,10 @@ void CCamera::OnRender()
 
 	if(!(m_pClient->m_Snap.m_SpecInfo.m_Active || IsRace(&Info) || IsBlockWorlds(&Info) || Client()->State() == IClient::STATE_DEMOPLAYBACK))
 	{
-		m_ZoomSet = false;
-		m_Zoom = 1.0;
+		if (!Client()->DummyConnected() && !Client()->DummyConnecting()) {
```
Comment (heinrich5991) [inline src/game/client/components/camera.cpp]:
```suggestion
		if(!Client()->DummyConnected() && !Client()->DummyConnecting())
		{
```

**Comment by archimede67** [comment]
Ok so what should we do now?

**Comment by def-** [comment]
I'm waiting for a third admin opinion. My vote is +1, jao's -1, so next one decides I guess.

**Comment by fokkonaut** [comment]
I wouldnt merge it.

**Comment by 12pm** [comment]
@def- +1 from me, I think having to adjust zoom on both tees separately is inconvenient for the casual player

**Comment by def-** [comment]
bors r+

> @def- just merge it :)
> I wouldnt merge it.

Give reasons or be quiet.

**Comment by fokkonaut** [comment]
b

**Pull request merged**